src/webview/memory-webview-view.tsx

```diff
@@ -222,78 +269,71 @@ class App extends React.Component<{}, MemoryAppState> {
                 showRadixPrefix={this.state.showRadixPrefix}
                 storeMemory={this.storeMemory}
                 applyMemory={this.applyMemory}
+                autoRefresh={this.state.autoRefresh}
+                autoRefreshDelay={this.state.autoRefreshDelay}
             />
         </PrimeReactProvider>;
     }
 
     protected updateMemoryState = (newState?: Partial<MemoryState>) => this.setState(prevState => ({ ...prevState, ...newState }));
     protected updateMemoryDisplayConfiguration = (newState: Partial<MemoryDisplayConfiguration>) => this.setState(prevState => ({ ...prevState, ...newState }));
     protected resetMemoryDisplayConfiguration = () => messenger.sendNotification(resetMemoryViewSettingsType, HOST_EXTENSION, undefined);
+
     protected updateTitle = (title: string) => {
         this.setState({ title });
         messenger.sendNotification(setTitleType, HOST_EXTENSION, title);
     };
 
     protected async setOptions(options?: MemoryOptions): Promise<void> {
         messenger.sendRequest(logMessageType, HOST_EXTENSION, `Setting options: ${JSON.stringify(options)}`);
-        if (this.state.configuredReadArguments.memoryReference === '') {
-            // Only update if we have no user configured read arguments
-            this.setState(prevState => ({ ...prevState, configuredReadArguments: { ...this.state.configuredReadArguments, ...options } }));
-        }
-
+        this.setState({ configuredReadArguments: { ...this.state.configuredReadArguments, ...options } });
         return this.fetchMemory(options);
     }
 
-    protected fetchMemory = async (partialOptions?: MemoryOptions): Promise<void> => this.doFetchMemory(partialOptions);
-    protected async doFetchMemory(partialOptions?: MemoryOptions): Promise<void> {
-        if (this.state.isFrozen) {
+    protected fetchMemory = async (partialOptions?: MemoryOptions): Promise<void> => {
+        if (this.state.isFrozen || !this.state.sessionContext.canRead) {
             return;
         }
-        this.setState(prev => ({ ...prev, isMemoryFetching: true }));
         const completeOptions = {
             memoryReference: partialOptions?.memoryReference || this.state.activeReadArguments.memoryReference,
             offset: partialOptions?.offset ?? this.state.activeReadArguments.offset,
             count: partialOptions?.count ?? this.state.activeReadArguments.count
         };
+        if (completeOptions.memoryReference === '') {
+            // may happen when we initialize empty
+            return;
+        }
+        this.doFetchMemory(completeOptions);
```

Comment:
```suggestion
        await this.doFetchMemory(completeOptions);
```

Reply:
I chose to return the `doFetchMemory` instead but if you prefer await, I'm also fine with that.